src/pages/admin/donate/components/bank-details-currencies/bank-details-currencies-config/BankDetailsCurrenciesConfig.ts

```diff
@@ -0,0 +1,68 @@
+import { ForwardRefExoticComponent, RefAttributes } from 'react';
+import { UahBankDetailsType, ForeignBankDetailsType, BankCurrency } from '../../../../../../types/admin/donate';
+import { GenericFormProps, GenericFormRef } from '../../generic-form/GenericForm';
+import { FieldValues } from 'react-hook-form';
+import { CorrespondentBankDetailsForm, createBankDetailsForm } from '../bank-details-factory/BankDetailsFactory';
+import { BankDetailsUahApi } from '../../../../../../services/api/admin/donate/bank-details-uah/bank-details-uah-api';
+import { ForeignBankDetailsApi } from '../../../../../../services/api/admin/donate/bank-details-foreign/bank-details-foreign-api';
+import { AxiosInstance } from 'axios';
+
+export interface BankDetailsConfig<TItem extends FieldValues> {
+    form: ForwardRefExoticComponent<GenericFormProps<TItem> & RefAttributes<GenericFormRef>>;
+    createEmptyItem: (data: Partial<TItem>) => TItem;
+    withCorrespondentBanks?: boolean;
+    fetch: (client: AxiosInstance, currency?: BankCurrency) => Promise<TItem[]>;
+    create: (client: AxiosInstance, data: Omit<TItem, 'id'>) => Promise<TItem>;
+    update: (client: AxiosInstance, id: number, data: Partial<TItem>) => Promise<TItem>;
+    delete: (client: AxiosInstance, id: number) => Promise<void>;
+    correspondentForm?: ForwardRefExoticComponent<GenericFormProps<any> & RefAttributes<GenericFormRef>>;
+    currency?: BankCurrency;
+}
+
+export const bankDetailsConfig: Record<string, BankDetailsConfig<any>> = {
```

Comment:
Maybe this can help to avoid `<any>`
```suggestion
export interface BankDetailsConfigMap {
    UAH: BankDetailsConfig<UahBankDetailsType>;
    USD: BankDetailsConfig<ForeignBankDetailsType>;
    EUR: BankDetailsConfig<ForeignBankDetailsType>;
}

export const bankDetailsConfig: BankDetailsConfigMap = {
```